src/Controller/Api/BookController.php

```diff
@@ -0,0 +1,139 @@
+<?php
+namespace App\Controller\Api;
+
+use App\Entity\Book;
+use App\Service\BookManager;
+use FOS\RestBundle\Controller\AbstractFOSRestController;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use FOS\RestBundle\Controller\Annotations\QueryParam;
+use FOS\RestBundle\Request\ParamFetcherInterface;
+use FOS\RestBundle\View\View;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+
+/**
+ * Class BookController
+ * @package App\Controller\Api
+ *
+ */
+class BookController extends AbstractFOSRestController
+{
+    private BookManager $bookManager;
+
+    public function __construct( BookManager $bookManager)
+    {
+        $this->bookManager = $bookManager;
+    }
+
+    /**
+     * @Rest\Get(path="/api/books")
+     * @Rest\View(serializerGroups={"Default"})
+     * @QueryParam(name="page", requirements="\d+", default="1", description="The page number")
+     * @QueryParam(name="name", default="", description="The name of the book")
+     * @QueryParam(name="genre", default="", description="The genre of the book")
+     *
+     * @param ParamFetcherInterface $paramFetcher
+     * @return View
+     */
+    public function getBooks(ParamFetcherInterface $paramFetcher): View
+    {
+        $pagination = $this->bookManager->getBooks($paramFetcher);
```

Comment:
And here you're passing this black box down the line, and your underlying domain layer code has to deal with something, that is out of its control. ParamFetcher can exhibit unexpected behavior and that will break the app.

Reply:
I totally understand this issue. I never thought about this when using ParamFetcher to get the input and now, you're telling me this makes sense. Still, wondering what should I use ? Get the Request object as parameter and pass it to the book manager ? In this case, I should get ride of all QueryParam annotations . Again, using the FosRest bundle was not a good idea